### ISSUE TITLE
error creaating a new site - git submodule checkout failing with npx blowfish tools

### ISSUE THREAD

**Issue by harryjduke** [nunocoracao/blowfish-tools#80]
Hi, I am using Windows 11 with PowerShell, I run `npx blowfish-tools` and select 'Setup a new website with Blowfish' and then get en error. Here is the log:

```
✔ Hugo is available
✔ Git is available
✔ Folder ok...
✔ Hugo site created
✔ Git initialized
✔ Blowfish installed
⠋ Configuring BlowfishError: ENOENT: no such file or directory, lstat 'C:\Users\Harry\Projects\personal\portfolio\repos\harryjduke.github.io\themes\blowfish\config\_default'
    at cpSyncFn (node:internal/fs/cp/cp-sync:56:13)
    at Object.cpSync (node:fs:3055:3)
    at utils.directoryCopy (file:///C:/Program%20Files/nvm/v23.0.0/node_modules/blowfish-tools/src/utils.js:156:8)
    at flow.configureNew (file:///C:/Program%20Files/nvm/v23.0.0/node_modules/blowfish-tools/src/flow.js:116:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'lstat',
  path: 'C:\\Users\\Harry\\Projects\\personal\\portfolio\\repos\\harryjduke.github.io\\themes\\blowfish\\config\\_default'
}
```

I noticed that this issue has a very similar error but it is not an internet or permissions issue.
#73

**Comment by thirstythurston** [comment]
Hello I have the exact same problem but in windows 10 when I run the blowfish-tools new mynewsite as per the blowfish installation manual.  I can manually copy over the files but the git submodule add commands don't actually add the files to the directory.  Can the installation manual be updated to expand on if the output of these issues is intentional or if there is a bug what is the workaround?

**Comment by pranavnasrani** [comment]
@thirstythurston @harryjduke 
i suspect blowfish-tools uses much the same process as the step-by-step installation, just automating it. 
btw, this issue is already addressed on the main repo. @schneekluth and @nunocoracao were very kind to address this.

According to the information you provided, you both are using Windows, and this bug only happens on windows.
this issue only exists on blowfish release v2.79.0 (which i suspect is what you are using) and only on Windows systems as that version (.79) had an example site that the Windows path system is unable to resolve, hence the git checkout failed, leaving config/_default empty as mentioned.

@thirstythurston this is exactly what you mentioned - the handbook had an unconventional path for Windows, and no, this is not intentional. Read on to fix this:


a few hours ago (20 hours), @nunocoracao fixed this bug and released blowfish v2.80.0

Try now; using `npx blowfish` this command should succeed (with all relevant customizations, provided you have prerequisites installed) since the bug is fixed.

if this does not work, try using the standard git submodule install, as this automated tool is relatively new. this should give the exact same results but you might be missing out on the automated tool to customise your website (can be done by manually modifying the .toml files - look at blowfish.page's excellent documentation). how to do this is in the fix summary.


### **Summary of fixes:** ###
`npx blowfish`
(should work)


if not then,

`hugo new site <name>`
`cd <name>`
`git init`
`git submodule add -b main https://github.com/nunocoracao/blowfish.git themes/blowfish`
`cd themes/blowfish`
`git checkout tags/v2.78.0`
`git switch -c site`


_git submodules is a rather longer method and @nunocoracao 's automation will do this all for you (most likely) now that the latest version of blowfish (.80) fixes the aforementioned error - still._

try one of these methods, and if your issue is still not resolved, this might be an independent issue - only on your machines.
let me know if this works. blowfish is a really great theme and this should enable you guys to use it (hopefully)

**Comment by thirstythurston** [comment]
Hello,

I forgot to update my comment but I did find a workaround and it was in part of the Hugo Quick installation instructions.  Those instructions had a warning that said don't use Windows Power Shell (WPS) or the Command Prompt (cmd).  So I ran the blowfish installation in Windows System Linux (WSL2) and everything worked fine and the local hugo server worked.  The colon in the path issue didn't mater even in WSL2 mounted windows directories.  While the warning is pretty clear, I stupidly missed it the first time around because hugo installs and passes the version check just fine in WPS and in cmd.   This isn't your problem per say (I stupid), but maybe add to the blowfish instructions a explicit recommendation for windows users to use WSL2 for their set up and installation.   I realize that may be not addressing the problem directly but it is an alternate solution.
